apps/eaas-api/src/app/controllers/auth.controller.ts

```diff
@@ -1,150 +1,87 @@
-      res.status(201).json({
-        user: dbUser,
-        tokens,
-      });
-    }
-  } catch (error) {
-    handleApiError(res, error);
+  if (!dbUser || !dbUser.id) {
+    res.status(400).json({ error: "could not create user" });
+  } else {
+    const tokens = await tokenService.generateAuthTokens(dbUser.id);
+    res.status(201).json({
+      user: dbUser,
+      tokens,
+    });
   }
 };
 
 const login = async (req: Request, res: Response) => {
   const { email, password } = req.body;
 
-  try {
-    const user = await authService.loginWithEmailAndPassword(email, password);
-    const tokens = await tokenService.generateAuthTokens(user.id);
+  const user = await authService.loginWithEmailAndPassword(email, password);
+  const tokens = await tokenService.generateAuthTokens(user.id);
 
-    res.status(200).json({ user, tokens });
-  } catch (error) {
-    handleApiError(res, error);
-  }
+  res.status(200).json({ user, tokens });
 };
 
 const logout = async (req: Request, res: Response) => {
   const { refreshToken } = req.body;
-  try {
-    await authService.logout(refreshToken);
-    res.status(204).send();
-  } catch (error) {
-    handleApiError(res, error);
-  }
+  await authService.logout(refreshToken);
+  res.status(204).send();
 };
 
 const refreshTokens = async (req: Request, res: Response) => {
-  try {
-    const { refreshToken } = req.body;
-    const tokens = await authService.refreshAuth(refreshToken);
-    res.status(200).json(tokens);
-  } catch (error) {
-    if (error.message.includes("invalid signature")) {
-      res.status(400).json({ error: error.message });
-    } else if (error.message.includes("token not found")) {
-      res.status(400).json({ error: error.message });
-    } else if (error.message.includes("jwt expired")) {
-      res.status(400).json({ error: "refresh token expired" });
-    } else {
-      res.status(400).json({ error: error.message });
-    }
-  }
+  const { refreshToken } = req.body;
+  const tokens = await authService.refreshAuth(refreshToken);
+  res.status(200).json(tokens);
 };
 
 const resetPassword = async (req: Request, res: Response) => {
-  try {
-    const { password, token } = req.body;
-    if (token) {
-      await authService.resetPassword(token.toString(), password);
-      res.status(204).send();
-    } else {
-      res.status(400).json({ error: "must include token" });
-    }
-  } catch (error) {
-    if (error.message.toString().includes("token not found")) {
-      res.status(400).json({ error: error.message });
-    } else if (error.message.toString().includes("jwt expired")) {
-      res.status(400).json({ error: "token expired" });
-    } else {
-      res.status(400).json({ error: error.message });
-    }
+  const { password, token } = req.body;
+  if (token) {
+    await authService.resetPassword(token.toString(), password);
+    res.status(204).send();
+  } else {
+    res.status(400).json({ error: "must include token" });
```

Comment:
In the spirit of thinking via middlewares, this seems like the kind of thing that might be good in a middleware.

Reply:
But I get it, there's nuance. Might not be worth the change, considering you do need to use `token` in `authService.resetPassword` anyway.